### PR TITLE
[SpankBangPlaylist] Add new extractor

### PR DESCRIPTION
**Pull request by JChris246**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### [SpankBangPlaylist] New extractor 

New extractor to pull spankbang playlists
Adjusted spankbang regex to recognize single playlist videos

### CODE REVIEW

File: youtube_dl/extractor/spankbang.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 
 class SpankBangIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:(?:www|m|[a-z]{2})\.)?spankbang\.com/(?P<id>[\da-z]+)/video'
+    _VALID_URL = r'https?://(?:(?:www|m|[a-z]{2})\.)?spankbang\.com/(?P<id>[\da-z-]+)/(?:video|playlist)'
```
Comment (dstftw) [inline youtube_dl/extractor/spankbang.py]: This should not match playlist URLs.

File: youtube_dl/extractor/spankbang.py [hunk 2]
```diff
@@ -94,3 +94,34 @@ def _real_extract(self, url):
...
+            'id': 'ug0k',
+            'title': 'Big Ass Titties playlist',
+        },
+        'playlist_mincount': 2,
+    }
+
+    def _extract_entries(self, webpage):
+        return [
+            self.url_result(
+                'http://www.%s/%s' % ('spankbang.com', video_url),
```
Comment (dstftw) [inline youtube_dl/extractor/spankbang.py]: `urljoin` or source `url`.

File: youtube_dl/extractor/spankbang.py [hunk 3]
```diff
@@ -94,3 +94,34 @@ def _real_extract(self, url):
...
+                SpankBangIE.ie_key())
+            for video_url in re.findall(
+                r'href="/?([\da-z-]+/playlist/[^"]+)', webpage)
```
Comment (dstftw) [inline youtube_dl/extractor/spankbang.py]: 1. This is too broad, regex should be restricted by playlist id.
2. This captures duplicates.

File: youtube_dl/extractor/spankbang.py [hunk 4]
```diff
@@ -94,3 +94,34 @@ def _real_extract(self, url):
...
+        ]
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        playlist_id = mobj.group('id')
+        webpage = self._download_webpage(url, playlist_id)
+
+        entries = self._extract_entries(webpage)
+        title = self._search_regex(r'<h1>(.+)</h1>', webpage, 'playlist_title')
```
Comment (dstftw) [inline youtube_dl/extractor/spankbang.py]: Playlist title should not be fatal.

File: youtube_dl/extractor/spankbang.py [hunk 5]
```diff
@@ -94,3 +94,34 @@ def _real_extract(self, url):
...
+            self.url_result(
+                'http://www.%s/%s' % ('spankbang.com', video_url),
+                SpankBangIE.ie_key())
+            for video_url in re.findall(
+                r'href="/?([\da-z-]+/playlist/[^"]+)', webpage)
+        ]
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        playlist_id = mobj.group('id')
```
Comment (dstftw) [inline youtube_dl/extractor/spankbang.py]: `_match_id`.

File: youtube_dl/extractor/spankbang.py [hunk 6]
```diff
@@ -94,3 +95,40 @@ def _real_extract(self, url):
...
+        video_items = re.findall(r'<div[^>]+class=[\'"].*?video-item[^>]*>\s*(.+?)>', webpage)
+
+        entries = []
+        if video_items:
+            for div in video_items:
+                page_url = self._search_regex(
+                    r'href="/?(' + id + '-[\da-z]+/playlist/[^"]+)', div, 'page url', default=None)
+
+                if page_url:
+                    page = self._download_webpage(urljoin('http://spankbang.com', page_url), id)
```
Comment (dstftw) [inline youtube_dl/extractor/spankbang.py]: No. It's a job if the video extractor.

File: youtube_dl/extractor/spankbang.py [hunk 7]
```diff
@@ -94,3 +95,40 @@ def _real_extract(self, url):
...
+    }
+
+    def _extract_entries(self, webpage, id):
+        video_items = re.findall(r'<div[^>]+class=[\'"].*?video-item[^>]*>\s*(.+?)>', webpage)
+
+        entries = []
+        if video_items:
+            for div in video_items:
+                page_url = self._search_regex(
+                    r'href="/?(' + id + '-[\da-z]+/playlist/[^"]+)', div, 'page url', default=None)
```
Comment (dstftw) [inline youtube_dl/extractor/spankbang.py]: Don't shadow built-in names.

File: youtube_dl/extractor/spankbang.py [hunk 8]
```diff
@@ -94,3 +95,40 @@ def _real_extract(self, url):
...
+            'title': 'Big Ass Titties playlist',
+        },
+        'playlist_mincount': 2,
+    }
+
+    def _extract_entries(self, webpage, id):
+        video_items = re.findall(r'<div[^>]+class=[\'"].*?video-item[^>]*>\s*(.+?)>', webpage)
+
+        entries = []
+        if video_items:
```
Comment (dstftw) [inline youtube_dl/extractor/spankbang.py]: This check is pointless.

File: youtube_dl/extractor/spankbang.py (hunk 6 above)
Comment (JChris246) [reply youtube_dl/extractor/spankbang.py]: This is needed to get the canonical url for the single video extractor [SpankBang] to use

Comment (dstftw) [reply youtube_dl/extractor/spankbang.py]: Again: video URLs are already available on playlist page.

Comment (JChris246) [reply youtube_dl/extractor/spankbang.py]: these urls are in the for /playlist_id-playlist_item_id/playlist/playlistname
even if you replace playlist with video and only use the playlist_item_id it does point to the the actual video url

Comment (JChris246) [reply youtube_dl/extractor/spankbang.py]: *does not

Comment (dstftw) [reply youtube_dl/extractor/spankbang.py]: I'm not talking about **these** URLs.

Comment (JChris246) [reply youtube_dl/extractor/spankbang.py]: Where are they

**Pull request closed without merging**